### ISSUE TITLE
Bug in multidimensional Akmaev adjustment

### ISSUE THREAD

**Issue by omiyawaki** [brian-rose/climlab#132]
The Akmaev scheme does not work with a multidimensional domain (my specific test case was a lat x lev space) because of issues in lines 28 and 50 in the akmaev_adjustment.py file. The issue begins with the shape of Pi as output from line 28:

Pi = np.cumprod((p / pextended[:-1])**alpha)

Here Pi takes the shape (lat*lev, 1) instead of (lat, lev) as it is supposed to. Second, in lines 50-51,

 theta[lat,:] = Akmaev_adjustment(theta[lat,:], q, beta, n_k,
                                              theta_k, s_k, t_k)

q and beta are also multi-dimensional arrays so Akmaev_adjusment() throws an error. For my specific domain that I was using, I fixed this by specifying the axis to take the cumulative product in line 28 as the one that corresponds to the vertical axis:

Pi = np.cumprod((p / pextended[:-1])**alpha, axis=1)

and specified the latitude for q and beta in line 50:

 theta[lat,:] = Akmaev_adjustment(theta[lat,:], q[lat,:], beta[lat,:], n_k,
                                              theta_k, s_k, t_k)

**Comment by brian-rose** [comment]
Thanks for this report! Can you post a complete minimal example that triggers the error?

**Comment by omiyawaki** [comment]
Here is a minimal example:

```python
import climlab

state = climlab.column_state(num_lev=3, num_lat=2)
model = climlab.TimeDependentProcess(state=state)
conv = climlab.convection.ConvectiveAdjustment(state=state, adj_lapse_rate='MALR')
model.add_subprocess('convection', conv)
model.integrate_days(1)
```

This gives me the error

```
ValueError: operands could not be broadcast together with shapes (2,4) (8,)
```

**Comment by brian-rose** [comment]
Curious. The problem goes away with a constant critical lapse rate:
```
import climlab
state = climlab.column_state(num_lev=3, num_lat=2)
conv = climlab.convection.ConvectiveAdjustment(state=state, adj_lapse_rate=6.5) 
conv.step_forward()
```
runs without error.

But this triggers the same error you describe:
```
import climlab
state = climlab.column_state(num_lev=3, num_lat=2)
conv = climlab.convection.ConvectiveAdjustment(state=state, adj_lapse_rate='MALR') 
conv.step_forward()
```

**Comment by brian-rose** [comment]
@omiyawaki Thanks for this bug report, the fix is incorporated into climlab 0.7.8, just released.
